### ISSUE TITLE
Abrupt pausing/black screen when using Fullscreen start/start screen in Windows 10

### ISSUE THREAD

**Issue by Neptune-mk2** [rocksdanister/lively#79]
**Before proceeding**
I have tried multiple things to try and prevent this from happening but I can't resolve it, therefor bug report.

**Describe the bug**
If you are using the Fullscreen Start option in Windows 10, opening fullscreen start with Lively running and right clicking on a tile/or oppening the app list in fullscreen start will result in lively pausing to a black screen.

**To Reproduce**
1. Open the settings app and enable the fullscreen start option in personalization>start
2. Open the start screen with lively running
3. Do any of the following actions : Right click on a tile, Open the applist, Open a start menu context menu, click on the background of the start screen
4. During any of these Lively will pause abruptly and then resume once focus is lost from the start screen

**Expected behavior**
Lively should not pause at any time when the start screen is open, applist open, or context menu is open, Lively should also not be pausing when you click on the start screen background once.

**Screenshots/Video**
N/A - Can't really record this or screenshot it

**Desktop (please complete the following information):**
 - OS: Windows 10 2004
- Any wallpaper

**Additional context**
I have successfully reproduced this on multiple desktop machines, a laptop, and a surface pro 1, all running Lively v0.9.6.0

**Log file**
Note - This bug does not cause a crash and does not seem to display any error at all, I am including the log file regardless
`2020-06-20 02:45:20.2686|INFO|livelywpf.SystemInfo|Lively v0.9.6.0 en-US  64Bit:False
2020-06-20 02:45:20.2884|INFO|livelywpf.SystemInfo|Portable build: False
2020-06-20 02:45:20.3785|INFO|livelywpf.SystemInfo|OS: Microsoft Windows 10 Pro 10.0.19041
2020-06-20 02:45:21.4484|INFO|livelywpf.SystemInfo|CPU: AMD Ryzen 7 1700 Eight-Core Processor          
2020-06-20 02:45:21.4737|INFO|livelywpf.SystemInfo|GPU: Radeon RX Vega
2020-06-20 02:45:22.6037|DEBUG|livelywpf.MainWindow|Single Display Mode:- Screen[Bounds={X=0,Y=0,Width=1920,Height=1080} WorkingArea={X=0,Y=0,Width=1920,Height=1040} Primary=True DeviceName=\\.\DISPLAY1
2020-06-20 02:45:22.6037|INFO|livelywpf.MainWindow|Display(s) wallpapers to load:-\\.\DISPLAY1
2020-06-20 02:45:22.6287|INFO|livelywpf.MainWindow|Restarting/Restoring All Wallpaper(s)
2020-06-20 02:45:22.6458|INFO|livelywpf.MainWindow|Setting up wallpaper:-C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\ekyf5ove.f5o\horizon.html
2020-06-20 02:45:23.2785|INFO|livelywpf.SetupDesktop|searching workerWOrig..
2020-06-20 02:45:23.2785|INFO|livelywpf.SetupDesktop|searching again shell_tray..
2020-06-20 02:45:23.2921|INFO|livelywpf.SetupDesktop|Cefsharp Handle:- HWND2163952
2020-06-20 02:45:23.2921|INFO|livelywpf.SetupDesktop|Sending WP -> Screen[Bounds={X=0,Y=0,Width=1920,Height=1080} WorkingArea={X=0,Y=0,Width=1920,Height=1040} Primary=True DeviceName=\\.\DISPLAY1
2020-06-20 02:45:23.3116|INFO|livelywpf.SetupDesktop|Relative Coordinates of WP -> 0 1920 1920 1080
2020-06-20 02:45:23.3116|INFO|livelywpf.SetupDesktop|Coordinate wrt to screen ->0 0 1920 1080
2020-06-20 02:45:35.3064|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\0lgotnzv.gks\Cyberpunk_2077_1080.mp4 video
2020-06-20 02:45:35.4779|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\1afxzb4m.fvj\ribbon.html web
2020-06-20 02:45:35.5149|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\242tny3g.eod\index.html web_audio
2020-06-20 02:45:35.5327|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\2elw0opr.keh\index.html web
2020-06-20 02:45:35.5508|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\bks1jbta.ez5\waves.html web
2020-06-20 02:45:35.5875|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\djcze13f.mlq\eternal_light_version2.mp4 video
2020-06-20 02:45:35.6150|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\ekyf5ove.f5o\horizon.html web
2020-06-20 02:45:35.6297|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\f1wdj2ap.hzk\outrun.html web
2020-06-20 02:45:35.6655|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\gyl1gr5r.hq4\waves.html web
2020-06-20 02:45:35.6798|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\gzmbwao1.2ii\index.html web
2020-06-20 02:45:35.6946|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\h33vsfxj.m4z\electric.html web
2020-06-20 02:45:35.7288|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\huryleus.43p\index.html web
2020-06-20 02:45:35.7598|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\lmzq3xt0.acw\starfield.html web
2020-06-20 02:45:35.7897|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\pavw10vh.lsx\index.html web
2020-06-20 02:45:35.8056|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\qny4he45.5zv\index.html web
2020-06-20 02:45:35.8334|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\s1qdxgqn.hhx\index.html web
2020-06-20 02:45:35.8730|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\tcgfexhe.evg\index.html web
2020-06-20 02:45:35.8992|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\vk1bqlvw.2ke\outrun.html web
2020-06-20 02:45:35.9141|INFO|livelywpf.MainWindow|Loading Wallpaper (wp dir):- C:\Users\Neptune\Documents\Lively Wallpaper\wallpapers\yxr5comn.hl2\fuji.html web
2020-06-20 02:45:35.9473|INFO|livelywpf.MainWindow|Loading Wallpaper (absolute path):- C:\Users\Neptune\Documents\Custom\Lively\custom_wallpapers\Neptune_Basilicom.mp4 video
2020-06-20 02:45:35.9473|INFO|livelywpf.MainWindow|Loading Wallpaper (absolute path):- C:\Users\Neptune\Documents\Custom\Lively\custom_wallpapers\XMB Waves in Purple.mp4 video
`

**Comment by rocksdanister** [comment]
Yes I am aware of this.
This should be resolved once I switch to checking UI classname instead of program name for excluding windows built in programs.

v1.0 should come with this change.

**Comment by Neptune-mk2** [comment]
> Yes I am aware of this.
> This should be resolved once I switch to checking classname instead of program name for excluding windows built in programs.

Ah good to hear, I will just close the issue for now since you are aware of the bug

**Comment by rocksdanister** [comment]
Nah keep it open, I'll add it as a v1.0 milestone.

**Comment by Neptune-mk2** [comment]
Alright will do 👍

**Comment by rocksdanister** [comment]
So I just had a look:

```
                const int maxChars = 256;
                StringBuilder className = new StringBuilder(maxChars);
                var handle = GetForegroundWindow();

                if (GetClassName((int)handle, className, maxChars) > 0)
                {
                    string cName = className.ToString();
                    Console.WriteLine(cName);
                    Debug.WriteLine(cName);
                }
                else
                {
                    try
                    {
                        int processID;
                        GetWindowThreadProcessId(handle, out processID);
                        var proc = Process.GetProcessById(processID);
                        Debug.WriteLine("Failed to get class:"+proc.ProcessName);
                    }
                    catch
                    {
                        Debug.WriteLine("getting processname failure, ignoring");
                        //ignore, admin process etc
                    }
                }
```
this is what the classes looks like:

```
Windows.UI.Core.CoreWindow = start menu, taskview, action center.. etc
ConsoleWindowClass = command prompt
MultitaskingViewFrame = alt+tab screen
Shell_TrayWnd = taskbar
ApplicationFrameWindow = uwp applications (win10 settings apps etc)
CabinetWClass, Explorer­WClass = windows explorer (Nice read: https://devblogs.microsoft.com/oldnewthing/20150619-00/?p=45341)
Chrome_widgetWin_1 = chrome and electron apps..
TaskManagerWindow = taskmanager
RainmeterMeterWindow = rainmeter widgets(?)

**Comment by rocksdanister** [comment]
Tested working, expect this fixed in v1.0 release.
